scalafix-tests/unit/src/test/scala/scalafix/tests/cli/SuppressTests.scala

```diff
@@ -0,0 +1,53 @@
+package scalafix.tests.cli
+
+import scala.meta.{Defn, Lit}
+import scalafix.lint.LintCategory
+import scalafix.rule.Rule
+
+object SuppressRules {
+  val NoVars: Rule = Rule.linter("NoVars") { ctx =>
+    val error = LintCategory.error("No vars!")
+    ctx.tree.collect {
+      case v @ Defn.Var(_, _, _, _) => error.at("no vars", v.pos)
+    }
+  }
+
+  val NoInts: Rule = Rule.linter("NoInts") { ctx =>
+    val error = LintCategory.error("No ints!")
+    ctx.tree.collect {
+      case i @ Lit.Int(_) => error.at("no ints", i.pos)
+    }
+  }
+}
+
+class SuppressTests extends BaseCliTest {
+  checkSuppress(
+    name = "suppress vars",
+    originalFile = """
+                     |object A {
+                     |  var a = 1
+                     |}
+                   """.stripMargin,
+    rule = "scala:scalafix.tests.cli.SuppressRules.NoVars",
+    expectedFile = """
+                     |object A {
+                     |  var/* scalafix:ok */ a = 1
```

Comment:
I'm a bit concerned about the strategy to place the comment right after the position of the reported message.  Firstly, I'm not sure how excited people would be to merge diffs like this. Secondly, I'm also not sure whether the comment will disable the given linter message in all cases.

I have a feeling it might be better to retrofit rules to support this mode. Something like `ctx.lint(LintMessage, autoSupresss = Patch): Patch` could do the job. Any thoughts @fommil ?

Reply:
Sounds good 👍 this generic approach should be OK then.